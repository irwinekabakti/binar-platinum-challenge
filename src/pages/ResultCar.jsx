import React from "react";
import Navbar from "../components/Navbar/Navbar";
import CardCar from "../components/CardCar/CardCar";
import Footer from "../components/Footer/Footer";
import Empty from "../components/Empty/Empty";
import SearchBar from "../components/SearchBar/SearchBar";

const ResultCar = () => {
  return (
    <div>
      <header>
        <Navbar />
        <Empty />
      </header>
      <main>
        <SearchBar />
        <CardCar />
      </main>
      <Footer />
    </div>
  );
};

export default ResultCar;
