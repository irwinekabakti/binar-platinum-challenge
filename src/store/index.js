import { configureStore } from "@reduxjs/toolkit";

import authSlice from "./action/action-slice";
import authAdminSlice from "./action/admin-slice";
import registerSlice from "./action/register-slice";

export const store = configureStore({
  reducer: {
    authStore: authSlice.reducer,
    regStore: registerSlice.reducer,
    adminStore: authAdminSlice.reducer,
  },
});
