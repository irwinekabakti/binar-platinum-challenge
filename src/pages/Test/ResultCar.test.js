import { BrowserRouter } from "react-router-dom";
import Renderer from "react-test-renderer";
import { describe, test, expect } from "@jest/globals";
import { Provider } from "react-redux";
import { store } from "../../store";
import ResultCar from "../ResultCar";

describe("UI ResultCar test", () => {
  test("UI ResultCar should match snapshot", () => {
    const snapshot = Renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <ResultCar />
        </BrowserRouter>
      </Provider>
    );

    expect(snapshot).toMatchSnapshot();
  });
});
