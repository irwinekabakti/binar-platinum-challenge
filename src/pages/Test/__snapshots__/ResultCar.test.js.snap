// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI ResultCar test UI ResultCar should match snapshot 1`] = `
<div>
  <header>
    <nav
      className="navbar navbar-expand-lg fixed-top navbarColor"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand navbarBrand1 nav-link"
          disabled={false}
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Home BCR"
            src="logo.svg"
          />
        </a>
        <button
          aria-controls="offcanvasNavbar"
          className="navbar-toggler navTog"
          data-bs-target="#offcanvasNavbar"
          data-bs-toggle="offcanvas"
          data-testid="btn-navbar-hamburger"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-labelledby="offcanvasNavbarLabel"
          className="offcanvas offcanvas-end"
          data-testid="toggle-navbar-hamburger"
          id="offcanvasNavbar"
          style={
            Object {
              "width": "50%",
            }
          }
          tabIndex="-1"
        >
          <div
            className="offcanvas-header"
          >
            <h5
              className="offcanvas-title fw-bold"
              id="offcanvasNavbarLabel"
            >
              BCR
            </h5>
            <button
              aria-label="Close"
              className="btn-close"
              data-bs-dismiss="offcanvas"
              type="button"
            />
          </div>
          <div
            className="offcanvas-body"
          >
            <ul
              className="navbar-nav justify-content-end flex-grow-1 pe-3"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link me-4 navbarNavLink"
                  href="/#our-services"
                >
                  Our Services
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link me-4 navbarNavLink"
                  href="/#WhyUs"
                >
                  Why Us
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link me-4 navbarNavLink"
                  href="/#testimonial"
                >
                  Testimonial
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link me-4 navbarNavLink"
                  href="/#faq"
                >
                  FAQ
                </a>
              </li>
              <button
                className="nav-link text-light me-4 mt-1 navbarRegister"
                data-testid="test-btnRegister"
                onClick={[Function]}
                style={
                  Object {
                    "border": "none",
                    "outline": "none",
                  }
                }
              >
                Register
              </button>
            </ul>
          </div>
        </div>
      </div>
    </nav>
    <section
      className="emptySection"
      data-testid="empty-Test"
      id="empty-section"
    />
  </header>
  <main>
    <section
      className="searchBar main"
      id="searchBar"
    >
      <div
        className="container mt-1 d-flex justify-content-center"
      >
        <div
          className="card p-4 cardBar"
        >
          <div
            className="row"
          >
            <p
              className="fs-6 fw-bold"
              style={
                Object {
                  "marginTop": "-20px",
                }
              }
            >
              Pencarianmu
            </p>
            <div
              className="col-lg-3 d-flex flex-column justify-content-between"
            >
              <label
                className="fw-bold labelCar"
                htmlFor="carName"
              >
                Nama Mobil
              </label>
              <input
                className="p-1 sizeTag"
                disabled={true}
                id="carName"
                name="carName"
                type="text"
              />
            </div>
            <div
              className="col-lg-3 d-flex flex-column justify-content-between"
            >
              <label
                className="fw-bold label"
                htmlFor="category"
              >
                Kategori
              </label>
              <select
                className="sizeTag"
                disabled={true}
                id="category"
                name="category"
              >
                <option
                  hidden={true}
                  value=""
                >
                  Masukkan Kapasitas Mobil
                </option>
              </select>
            </div>
            <div
              className="col-lg-3 d-flex flex-column justify-content-between"
            >
              <label
                className="fw-bold label"
                htmlFor="price"
              >
                Harga
              </label>
              <select
                className="sizeTag"
                disabled={true}
                id="price"
                name="price"
              >
                <option
                  hidden={true}
                  value=""
                >
                  Masukkan Harga Sewa per Hari
                </option>
              </select>
            </div>
            <div
              className="col-lg-2 d-flex flex-column justify-content-between"
            >
              <label
                className="fw-bold label"
                htmlFor="status"
              >
                Status
              </label>
              <select
                className="p-1 sizeTag"
                disabled={true}
                id="status"
                name="status"
              >
                <option
                  value="sedia"
                >
                  Sedia
                </option>
              </select>
            </div>
            <div
              className="col-lg-1 d-flex align-items-end mt-3"
            >
              <button
                className="btn btn-outline-primary w-100 col-sm-12 mt-4"
                onClick={[Function]}
              >
                Edit
              </button>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      className="container containerBottom"
      id="card-car"
    >
      <div
        className="row justify-content-evenly"
        data-testid="container-Content"
      >
        <div
          className="containerEmptyData d-flex justify-content-center mt-3"
        >
          <h3
            className="titleEmptyData fw-bold"
          >
            Ups, Sorry ! Data is Empty
          </h3>
        </div>
        <div
          className="containerDescription d-flex justify-content-center"
        >
          <h6
            className="titleDescription"
          >
            Please try again !
          </h6>
        </div>
      </div>
    </section>
  </main>
  <footer
    className="footer"
  >
    <div
      className="container py-5 pt-0"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-3 mb-3"
        >
          <ul
            className="list-unstyled"
          >
            <li
              className="mb-2"
            >
              <p
                className="fw-bold"
              >
                Jalan Suroyo No. 161 Mayangan Kota Probolonggo 672000
              </p>
            </li>
            <li
              className="mb-2"
            >
              <p
                className="fw-bold"
              >
                binarcarrental@gmail.com
              </p>
            </li>
            <li
              className="mb-2"
            >
              <p
                className="fw-bold"
              >
                081-233-334-808
              </p>
            </li>
          </ul>
        </div>
        <div
          className="col-lg-2 mb-3"
        >
          <ul
            className="list-unstyled"
          >
            <li
              className="mb-2 listLink"
            >
              <a
                className="text-decoration-none linkFooter"
                href="/#services"
              >
                Our services
              </a>
            </li>
            <li
              className="mb-2 listLink"
            >
              <a
                className="text-decoration-none linkFooter"
                href="/#WhyUs"
              >
                Why Us
              </a>
            </li>
            <li
              className="mb-2 listLink"
            >
              <a
                className="text-decoration-none linkFooter"
                href="/#testimonial"
              >
                Testimonial
              </a>
            </li>
            <li
              className="mb-2 listLink"
            >
              <a
                className="text-decoration-none linkFooter"
                href="/#faq"
              >
                FAQ
              </a>
            </li>
            <li
              className="mb-2 listLink"
            >
              <a
                className="text-decoration-none linkFooter"
                href="/viewerPdf"
              >
                Term of Service
              </a>
            </li>
          </ul>
        </div>
        <div
          className="col-lg-4 mb-3"
        >
          <p
            className="fw-bold mb-2 "
          >
            Connect with us
          </p>
          <div
            className="footerSocial"
          >
            <a
              className="text-decoration-none footerSocial"
              href="https://web.facebook.com/binaracademy/"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="icon-facebook"
                data-bs-placement="top"
                data-bs-toggle="tooltip"
                src="icon_facebook.svg"
                title="Facebook"
              />
            </a>
            <a
              className="text-decoration-none footerSocial"
              href="https://www.instagram.com/academybinar/"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="icon_instagram"
                data-bs-placement="top"
                data-bs-toggle="tooltip"
                src="icon_instagram.svg"
                title="Instagram"
              />
            </a>
            <a
              className="text-decoration-none footerSocial"
              href="https://twitter.com/academybinar"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="icon_twitter"
                data-bs-placement="top"
                data-bs-toggle="tooltip"
                src="icon_twitter.svg"
                title="Twitter"
              />
            </a>
            <a
              className="text-decoration-none footerSocial"
              href="mailto:service@binar.co.id"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="icon_gmail"
                data-bs-placement="top"
                data-bs-toggle="tooltip"
                src="icon_mail.svg"
                title="Gmail"
              />
            </a>
            <a
              className="text-decoration-none footerSocial"
              href="https://www.binaracademy.com/"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="icon_twitch"
                data-bs-placement="top"
                data-bs-toggle="tooltip"
                src="icon_twitch.svg"
                title="Twitch"
              />
            </a>
          </div>
        </div>
        <div
          className="col-lg-3 mb-3"
          data-testid="contain-LogoFooter"
        >
          <p
            className="fw-bold mb-2"
            data-testid="paragraph-Footer"
          >
            Copyright Binar 2022
          </p>
          <a
            className="navbar-brand"
            href="/"
          >
            <img
              alt="Home BCR"
              src="logo.svg"
            />
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
